### ISSUE TITLE
The update process aborts with v5.2.26-2

### ISSUE THREAD

**Issue by lfield** [EGI-Foundation/bdii#38]
The BDII was updated to v5.2.26-2 on 22nd of Jan. The update process aborts with the following error. 
```
Traceback (most recent call last):
  File "/usr/sbin/bdii-update", line 1031, in <module>
    main(config, log)
  File "/usr/sbin/bdii-update", line 804, in main
    for dn in delay_delete.keys():
RuntimeError: dictionary changed size during iteration
```
Please don't push new packages to production on Friday afternoon!

**Comment by andrea-manzi** [comment]
Hi @lfield ,
this comes from the EPEL update pushed by Matthias which switched to Py3 also on EPEL7

https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2021-2978244eeb

i tried to ask  there not to apply this change ( also cause we didn't do proper testing on Py3 yet) but i got no reply

@ellert  can we move back to py2 on EPEL7 now as there are more issues to solve on the py3 version?

thanks
Andrea

**Comment by lfield** [comment]
The other issue is that I can't even downgrade as the previous version has been deleted.

**Comment by andrea-manzi** [comment]
you can use the UMD repo version of the package

https://repository.egi.eu/sw/production/umd/4/centos7/x86_64/updates/

we tried to align what we have released there with EPEL7 but so far we failed

**Comment by ellert** [comment]
https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2021-7039c96c81

**Comment by ellert** [comment]
Please add karma if you want it to migrate faster.

(If you need old builds they are available in koji: https://koji.fedoraproject.org/koji/packageinfo?packageID=10417)

**Comment by lfield** [comment]
I prioritized the UMD repository and downgraded.

**Comment by andrea-manzi** [comment]
thanks  a lot @ellert i have added karma to the package ( so it's already being pushed to stable)

**Comment by gwarf** [comment]
Closed by #39
